app/controllers/document_lead_image_controller.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class DocumentLeadImageController < ApplicationController
+  def index
+    @document = Document.find_by_param(params[:document_id])
+  end
+
+  def create
+    document = Document.find_by_param(params[:document_id])
+    upload = UploadedImageService.new(params.require(:image)).process
+
+    if upload.valid?
+      image = create_image_from_upload(upload, document)
+      redirect_to edit_document_lead_image_path(params[:document_id], image.id)
+    else
+      redirect_to document_lead_image_path, alert: upload.errors
+    end
+  end
+
+  def edit
+    @document = Document.find_by_param(params[:document_id])
+    @image = Image.find_by(id: params[:image_id])
+  end
+
+  def update
+    document = Document.find_by_param(params[:document_id])
+    image = Image.find_by(id: params[:image_id])
+    image.update(update_params)
+    document.update(lead_image_id: image.id)
+    if params[:next] == "lead-image"
```

Comment:
This is a bit strange - why is the key called ':next'?

Reply:
Ah, this is quite confusing/unusual. I think this should be split into two actions, especially as one of the ways this is called doesn't actually update the image attributes.

Reply:
it's a difference between if you're editing the metadata for an existing image (returns to lead-image page) or you're uploading a new lead image (returns to document summary).